core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -52,17 +52,27 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
    */
   val initial: ZIO[R, Nothing, State]
 
+  /**
+   * Extract the B from the schedule
+   */
+  val extract: (A, State) => B
+
   /**
    * Updates the schedule based on a new input and the current state.
    */
-  val update: (A, State) => ZIO[R, Nothing, ZSchedule.Decision[State, B]]
+  val update: (A, State) => ZIO[R, Unit, State]
 
   /**
    * Returns a new schedule that continues only as long as both schedules
    * continue, using the maximum of the delays of the two schedules.
    */
   final def &&[R1 <: R, A1 <: A, C](that: ZSchedule[R1, A1, C]): ZSchedule[R1, A1, (B, C)] =
-    combineWith(that)(_ && _, _ max _)
+    new ZSchedule[R1, A1, (B, C)] {
+      type State = (self.State, that.State)
+      val initial = self.initial.zip(that.initial)
```

Comment:
I think we want to `zipPar` the initial states, too.

Reply:
👍